lib/datadog/di/component.rb

```diff
@@ -16,22 +16,22 @@ module DI
     # resources and installed tracepoints upon shutdown.
     class Component
       class << self
-        def build(settings, agent_settings, telemetry: nil)
+        def build(settings, agent_settings, logger, telemetry: nil)
```

Comment:
Minor: May be worth moving to all keyword arguments? (Although the type checker will complain if we get this wrong since you added signatures -- nice!)